src/fhirtypes/common.ts

```diff
@@ -198,6 +198,21 @@ export function setPropertyOnInstance(
           if (current[key] == null) current[key] = {};
           current = current[key];
         } else {
+          // Check if the instance already has the element being defined
+          if (current[pathPart.base] != null && typeof current[pathPart.base] === 'object') {
```

Comment:
I just noticed this now, but I'm wondering... why do you use `current[pathPart.base]` here (and after) rather than `current[key]`?  I only noticed this because on line 211, we essentially assign `assignedValue.value` to `current[pathPart.base].value`, then on line 216, we assign `current[key] = assignedValue`.  So if you follow along, we say we're preserving the `value` but we're basically doing `current[key].value = current[pathPart.base].value`; which really should only work if `key === pathPart.base`.

And sure enough, if you look up at lines 136-137 we have:
```ts
const key =
  pathPart.primitive && i < pathParts.length - 1 ? `_${pathPart.base}` : pathPart.base;
```
And we only get to your new code if `i === pathParts.length` so I think we do in fact know that `key === pathPart.base` 100% of the time we are in this block (lines 201-215).  So... to me, it would be more clear if we just used `current[key]` instead of `current[pathPart.base]` since the code directly above and below this uses `current[key]` already.

Reply:
You're right! Using `pathPart.base` isn't necessary when the `key` value is set to be the same thing on lines 136-137. There's no particular reason for my not using the `key` variable here, I had just blanked on the fact that they should always be equal in this block of code. I'll update the conditionals on lines 202 and below to use the `key` variable.